### DISCUSSION TITLE
My map files are corrupted (it's a me-problem)

### DISCUSSION THREAD

**Discussion by Lokizypresse902** [ppy/osu#22419] [Q&A]
### Type

Game behaviour

### Bug description

So I wasn't sure if I really wanted to submit a bug report since I often switch between stable and lazer but it's bothering me so I wanted to ask for help:

I wanted to move my osu!lazer folder to another location and CUT and pasted the whole thing intead of COPY and pasting it.
Silly how I am, I aborted the process of cutting and pasting it in the mid and because windows is just as silly, half of the maps got corrupted. That was several months ago.

Now, they still show in song select: I see their name and stats but sometimes there's no music playing, all the difficulties are grayed out (have 0.00 stars), it shows the map is 0 seconds long yada yada
I can't play them.

I could theoretically delete and download all maps manually but that's pretty tedious
Although that doesn't always work. I often only get the music playing back. (there's still no map to play)
Reimporting all the maps from stable back into lazer fixes it as well: It's just that I have a lot of maps that I downloaded on lazer but not on stable, so those remain unaffected
However, I imagine this is somehow possible to fix without losing all my maps because the map ID's are theoretically still there?

### Version

2023.123.0

### Logs

[legacy-ipc.log](https://github.com/ppy/osu/files/10501172/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/10501173/network.log)
[performance.log](https://github.com/ppy/osu/files/10501174/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10501175/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10501178/updater.log)
[database.log](https://github.com/ppy/osu/files/10501179/database.log)

**Comment by bdach** [comment]
Unfortunately, there is currently no existing recovery process that is immediately usable in such a case. Depending on your technical skills you may be able to conjure one, but it would not be trivial.

>  Although that doesn't always work. I often only get the music playing back. (there's still no map to play)

You may need to close and reopen the game after deleting maps to get this to work. In order to support restore, maps are *actually* purged on next game restart.

---

If you ever need to move your game folder again, there is an in-game facility to do this (settings -> *Change folder location...*)

**Reply by Lokizypresse902** [reply]
Thanks for the fast reply - yeah if I restart the game then redownloading works
" there is an in-game facility to do this "
that's... convenient. lol.
guess I'll try redownloading some maps
thanks again

**Comment by peppy** [comment]
Just to confirm, you deleted the previous directory after only half-copying the files?

**Reply by Lokizypresse902** [reply]
I deleted the pasted folder since it was incomplete and I could not even start the game.
I'm using the original folder that lazer was installed in.
But presuming windows copies, pastes and deletes files if I cut and paste them, then there should be a lot of files missing in that folder.
Reading that, I realized it might be fixable if the files I copied still existed by maybe just copying them back. well rip, they don't exist
